includes/manage_collection.inc

```diff
@@ -328,7 +333,16 @@ function islandora_basic_collection_policy_management_form_submit(array $form, a
     $content_model = islandora_object_load($pid);
     $policy->addContentModel($pid, $properties['prompt'], $properties['namespace']);
   }
-  $collection['COLLECTION_POLICY']->content = $policy->getXML();
+  if (!isset($collection['COLLECTION_POLICY'])) {
+    $cp_ds = $collection->constructDatastream('COLLECTION_POLICY', 'M');
```

Comment:
I've just seen a trend towards creating XML as managed is all - I believe the reasoning is that Fedora 3.x doesn't handle modifications to XML very well when multithreading (e.g., if something was modifying the RELS and something else was modifying the COLLECTION_POLICY and they were both inline, some fancy 409 errors might crop up). Rare issue but definitely happens when batching multiple processes simultaneously.